diagrams/entrances/diagraminfo/entrance.ftl

```diff
@@ -2,34 +2,53 @@
 
 [#macro diagrams_entrance_diagraminfo ]
 
-  [#-- override the deployment group to get all deployment groups --]
-  [@addCommandLineOption
-      option={
-        "Deployment" : {
-          "Framework" : {
-            "Name" : DIAGRAMS_EXEC_DEPLOYMENT_FRAMEWORK
-          },
-          "Provider" : {
-            "Names" : combineEntities(
-                        [ "diagrams" ],
-                        commandLineOptions.Deployment.Provider.Names,
-                        UNIQUE_COMBINE_BEHAVIOUR
-                      )
-          }
-        },
-        "View" : {
-          "Name" : DIAGRAMINFO_VIEW_TYPE
-        },
-        "Flow" : {
-          "Names" : [ "views" ]
-        }
-      }
+  [@generateOutput
+    deploymentFramework=getCLODeploymentFramework()
+    type=getCLODeploymentOutputType()
+    format=getCLODeploymentOutputFormat()
   /]
 
-  [@generateOutput
-    deploymentFramework=commandLineOptions.Deployment.Framework.Name
-    type=commandLineOptions.Deployment.Output.Type
-    format=commandLineOptions.Deployment.Output.Format
+[/#macro]
+
+[#macro diagrams_entrance_diagraminfo_inputsteps ]
+
+  [@registerInputSeeder
+    id=DIAGRAMINFO_ENTRANCE_TYPE
+    description="Entrance"
+  /]
+
+  [@addSeederToConfigPipeline
+    stage=COMMANDLINEOPTIONS_SHARED_INPUT_STAGE
+    seeder=DIAGRAMINFO_ENTRANCE_TYPE
   /]
 
 [/#macro]
+
+[#function diagraminfo_configseeder_commandlineoptions filter state ]
+  [#return
+    mergeObjects(
```

Comment:
See comment above re function to add to the input state

Reply:
Thanks @ml019 , updated both calls in https://github.com/hamlet-io/engine-plugin-diagrams/pull/16/commits/9d83362b3d42bdec4222a60685d375c71d7a4c1e